### PR TITLE
codelite 11.0 (new formula)

### PR DESCRIPTION
**Pull request by s172262**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

**Comment by DomT4** [comment]
```
Error: 2 problems in 1 formula
codelite:
  * C: 2: col 9: Description should start with a capital letter
  * C: 26: col 1: Use 2 spaces for indentation in a heredoc by using `<<~` instead of `<<~`.
```

File: Formula/codelite.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  def install
+    # https://github.com/eranif/codelite/issues/1284
+    # https://github.com/eranif/codelite/pull/1797
+    inreplace "CMakeLists.txt", "set(WX_COMPONENTS \"std\")", "set(WX_COMPONENTS \"std aui propgrid ribbon\")"
+    inreplace "codelite_terminal/CMakeLists.txt", "wxWidgets REQUIRED", "wxWidgets COMPONENTS std stc REQUIRED"
+    mkdir "build-release" do
+      system "cmake", "..", "-DCL_PREFIX=#{prefix}", *std_cmake_args
+      system "make", "install"
+      mkdir prefix/"libexec"
```
Comment (DomT4) [inline Formula/codelite.rb]: Why?

File: Formula/codelite.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      prefix.install "codelite.app"
+    end
+  end
+
+  def caveats; <<~EOS
+      Known Issue:
+      If you built `wxmac` with "--with-stl" option, you cannot build this Formula.
```
Comment (DomT4) [inline Formula/codelite.rb]: Given most people don't reach for `brew info` to resolve curiosity about new formulae, it's likely by the time most people sees this either they'll have poured the bottle & it'll be irrelevant, or their build will have blown up and they'll go looking for why. I'm not sure this provides value as-is, personally.

File: Formula/codelite.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  depends_on "wxmac"
+
+  def install
+    # https://github.com/eranif/codelite/issues/1284
+    # https://github.com/eranif/codelite/pull/1797
+    inreplace "CMakeLists.txt", "set(WX_COMPONENTS \"std\")", "set(WX_COMPONENTS \"std aui propgrid ribbon\")"
+    inreplace "codelite_terminal/CMakeLists.txt", "wxWidgets REQUIRED", "wxWidgets COMPONENTS std stc REQUIRED"
+    mkdir "build-release" do
+      system "cmake", "..", "-DCL_PREFIX=#{prefix}", *std_cmake_args
+      system "make", "install"
```
Comment (DomT4) [inline Formula/codelite.rb]: What is this installing if you're having to do a manual `prefix.install` below?

File: Formula/codelite.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+  def caveats; <<~EOS
+      Known Issue:
+      If you built `wxmac` with "--with-stl" option, you cannot build this Formula.
+      Because this Formula requires cxx11.
+    EOS
+  end
+
+  test do
+    system "#{opt_prefix}codelite.app/Contents/MacOS/codelite", "--version"
```
Comment (DomT4) [inline Formula/codelite.rb]: `opt_prefix` doesn't chuck in a trailing `/` so you'll want to add one, or this path will end up being read as:
```
/usr/local/opt/codelitecodelite.app/Contents/MacOS/codelite
```
I believe.

File: Formula/codelite.rb (hunk 3 above)
Comment (s172262) [reply Formula/codelite.rb]: `"make", "install"` makes install products and all dependency libraries into "#{buildpath}/build-release/codelite.app/Contents" for building `app`.

**Comment by ilovezfs** [comment]
Going to pass on this as it's building a .app and belongs in Cask, but thanks nonetheless @s172262!

**Pull request closed without merging**